tests/spec/06-mqtt-client_spec.lua

```diff
@@ -99,17 +101,6 @@ describe("MQTT client", function()
 				username = "stPwSVV73Eqw5LSv0iMXbc4EguS7JyuZR9lxU5uLxI5tiNM8ToTVqNpu85pFtJv9",
 			}
 		},
-		{
```

Comment:
Why this test case removed?

Reply:
the "sync" loop was broken and should not be used. It might even be dangerous, it could hang copas in a spin (when a connection fails and there is an auto-reconnect). So I removed the sync loop from `init.lua`, and hence had to remove the tests as well (just fixed all the tests, see the last one, the copas one for a correct implementation). 

Essentially the different loops (copas, nginx) require their own set up, they are too different wrt loops, timers, sleeping, etc. The sync loop was also too limited as it didn't do keepalives etc.

I was thinking of maybe adding module for copas and nginx with add/remove methods, such that
- `ioloop:add(...), copas:add(...), nginx:add(...)`, and
- `ioloop:remove(...), copas:remove(...), nginx:remove(...)`

would be identical in function signatures, just implementing it in different environments. It would actually be pretty easy to detect the environment automatically.

Reply:
Sync loop can be used in the simplest mqtt case - connect, subscribe, publish command, receive answer, disconnect.
This zero-depenndence scenario have to be preserved.

Reply:
Even in mqtt docs described simpler case - connect, publish, disconnect. Even without CONNACK check This is a typical behavior for IoT sensors to periodically publish it's value

See "Non normative comment" above http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/errata01/os/mqtt-v3.1.1-errata01-os-complete.html#_Toc442180846